dist: xenial
sudo: true
language: python

os:
  - linux

addons:
  apt:
    sources:
      - ubuntu-toolchain-r-test
    packages:
      - gcc-4.9
      - g++-4.9

python:
  - 2.7
  - 3.5
  - 3.6
  - 3.7

# Test all Python version with the latest PyTorch.
env:
  - TORCH_PIP_PACKAGE=torch

# Test other PyTorch/Python combinations.
matrix:
  include:
    - python: 3.6
      env: TORCH_PIP_PACKAGE="torch==1.0.0"
    - python: 3.6
      env: TORCH_PIP_PACKAGE="torch==1.0.1"

before_install:
  # Upgrade PIP to latest version, in order to support --progres-bar
  - pip install -U pip
  # Manually install torch, and pybind11 since they are required in the setup
  - pip install numpy --progress-bar off
  - pip install $TORCH_PIP_PACKAGE --progress-bar off
  - pip install pybind11 --progress-bar off

install:
  - export CC=gcc-4.9
  - export CXX=g++-4.9
  - python setup.py install

script:
  - cd /tmp
  - python -m unittest torch_baidu_ctc.test

notifications:
  email:
    on_success: change
    on_failure: always

# TODO(jpuigcerver): This fails: "Invalid or non-existent authentification".
#deploy:
#  provider: pypi
#  user: joapuipe
#  password:
#    secure: Eg2bJZQIh+XPVm6eObMiNu+YtgrwHlNQzddnhVBGBkF9E7WZlPHt6gvDsYkVVjrJjjH78LLWDtc5UZoCcyA2+HVIpr9Xpt3jNAvnVEf4N6vP5VuCCsw4foGbjHe9GLc60KNSe9Sx7Sd7tMtoW6Mph5JNbbUY3SThLvQ1mI/A38SldzIWjO8ndaw5YOp097I+eF1pAPczB8apYFlBCmbqiujs2yE1phcz19/sgn6nYZBooUGP3YjvtveLXyBKbMnjJ3ERUOBFBreBn6MCCmQxyHkoDXqJEbtAZ9KFJ0+3f7xUeLzWsWJREHKQWO4qzqzJT+6M8UAx6zq23xjDWdWmCek58AgIjRzG2YjibbnSE1OXKQduNdSR2DDfQP1nRtLooYjIzyz48LDAv2E2lUqx/0KkHonNk3vyIsDRR8NOYw+2kiIX1SJSH/Q/OW3pSuoO1/wIlgLRG70mGuOS3H3ok1JNGLyzpHJobTzzFFFJ8zPpGTgpTibbYcuTsJXLBIZO8xSHSTEG5jSCqtu8LrdIMqfPhwrfJxLEXBVvgah1eE30yOmOsbwNCfhLM31ebFWatbqCjlW4R9MG/2csYbamJp16ghd0NkGsSnkuxyvVG2h1qlutcPTCihOg+fReuZdo3/oFzhxiHfRsolE/scsHNGqEoS9xG+oywK5d0pv2CyM=
#  distributions: "sdist"
#  on:
#    tags: true
